hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/HiveSyncConfig.java

```diff
@@ -129,6 +129,8 @@ public static class HiveSyncConfigParams {
     public Boolean createManagedTable;
     @Parameter(names = {"--batch-sync-num"}, description = "The number of partitions one batch when synchronous partitions to hive")
     public Integer batchSyncNum;
+    @Parameter(names = {"--partition-cascade"}, description = "Partition cascade when table columns change.")
+    public Boolean partitionCascade;
```

Comment:
in general we should avoid adding parameters/configs to tweak the logic inside the sync. There are so many configs already exposing the impl. details and making meta sync hard to use. 

For this logic of determining cascade, can you figure out a way to detect it from schema/column changes?

Reply:
Hi, good idea about the configs settings.

And the logic cascade could not be detected from schema/column changes as far as I think, but the **table parameters** might help set this config in my opinion, then the user needs to create/alter the table with TBLPROPERTIES: "hoodie.datasource.hive_sync.partition_cascade=false"  through HiveSQL,  and this is not directly to use.

If we compare these two plans, I prefer the original plan, then what's your opinion?